### ISSUE TITLE
Add a GitHub action check on commit message

### ISSUE THREAD

**Issue by rbradford** [cloud-hypervisor/cloud-hypervisor#5840]
Add a (not required) check that the commit message body is appropriately formatted  to avoid commits like https://github.com/cloud-hypervisor/cloud-hypervisor/commit/7f47a030e72d7aca6897f433b156044ad3da6545 getting past review.

This should be warning only as there are many legitimate reasons why the commit message might need to be longer e.g. clippy reported errors - https://github.com/cloud-hypervisor/cloud-hypervisor/commit/239f422203306b1423d29322067ab133e4e106dd

**Comment by likebreath** [comment]
Oops, I missed the formatting issue on that commit body.

**Comment by likebreath** [comment]
@rmstoi Are you interested to take this issue? Thanks.

**Comment by rmstoi** [comment]
@likebreath I assigned this issue to myself.

@rbradford 
What is meant here by message format? Is the target to verity that commit messages adhere to 50/72 style, basically checking the length alone?

**Comment by likebreath** [comment]
> What is meant here by message format? Is the target to verity that commit messages adhere to 50/72 style, basically checking the length alone?

Yes, that's right. It would also be good to add checks for `<component>: Change summary` as described here: https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/CONTRIBUTING.md#patch-format

**Comment by rmstoi** [comment]
Is it needed to check component names for sanity?
If yes, is there a list of valid component names?

**Comment by likebreath** [comment]
We can start with the following component list:

```
api_client
arch
block
build
ch-remote
ci
devices
docs
event_monitor
fuzz
github
gitignore
hypervisor
Jenkinsfile
misc
net_gen
net_util
option_parser
pci
performance-metrics
rate_limiter
README
resources
scripts
serial_buffer
test_data
test_infra
tests
tpm
tracer
vhost_user_block
vhost_user_net
virtio-devices
vm-allocator
vm-device
vmm
vm-migration
vm-virtio
```
